File: library/src/main/java/me/proxer/library/api/manga/MangaApi.java
package me.proxer.library.api.manga;

import me.proxer.library.enums.Language;
import retrofit2.Retrofit;

/**
 * API for the Manga class.
 *
 * @author Ruben Gees
 */
public final class MangaApi {

    private final InternalApi internalApi;

    /**
     * Only for internal use.
     */
    public MangaApi(final Retrofit retrofit) {
        this.internalApi = retrofit.create(InternalApi.class);
    }

    /**
     * Returns the respective endpoint.
     */
    public ChapterEndpoint chapter(final String entryId, final int episode, final Language language) {
        return new ChapterEndpoint(internalApi, entryId, episode, language);
    }
}

File: library/src/main/kotlin/me/proxer/library/api/manga/MangaApi.kt
package me.proxer.library.api.manga

import me.proxer.library.enums.Language
import retrofit2.Retrofit

/**
 * API for the Manga class.
 *
 * @author Ruben Gees
 */
class MangaApi internal constructor(retrofit: Retrofit) {

    private val internalApi = retrofit.create(InternalApi::class.java)

    /**
     * Returns the respective endpoint.
     */
    fun chapter(entryId: String, episode: Int, language: Language): ChapterEndpoint {
        return ChapterEndpoint(internalApi, entryId, episode, language)
    }
}
